src/main/flush.ts

```diff
@@ -0,0 +1,166 @@
+import fs from 'fs';
+
+export enum FilterTypeEnum {
+  PEAK = 'PK', // ["PK",True,True]
+  LPQ = 'LPQ', // ["LPQ",False,True]
+  HPQ = 'HPQ', // ["HPQ",False,True]
+  BP = 'BP', // ["BP",False,True]
+  LS = 'LS', // ["LS",True,False]
+  HS = 'HS', // ["HS",True,False]
+  NO = 'NO', // ["NO",False,True]
+  AP = 'AP', // ["AP",False,True]
+  LSC = 'LSC', // ["LSC",True,True]
+  HSC = 'HSC', // ["HSC",True,True]
+  BWLP = 'BWLP', // ["BWLP",False,True]
+  BWHP = 'BWHP', // ["BWHP",False,True]
+  LRLP = 'LRLP', // ["LRLP",False,True]
+  LRHP = 'LRHP', // ["LRHP",False,True]
+  LSCQ = 'LSCQ', // ["LSCQ",True,True]
+  HSCQ = 'HSCQ', // ["HSCQ",True,True]
+}
+
+interface IFilter {
+  frequency: number;
+  gain: number;
+  type: FilterTypeEnum;
+  quality: number;
+}
+
+export interface IState {
+  isEnabled: boolean;
+  preAmp: number;
+  filters: IFilter[];
+}
+
+const FIXED_FREQUENCIES = [
+  32, 64, 125, 250, 500, 1000, 2000, 4000, 8000, 16000,
+];
+
+export const DEFAULT_FILTER: IFilter = {
+  frequency: 1000,
+  gain: 0,
+  quality: 1,
+  type: FilterTypeEnum.PEAK,
+};
+
+const DEFAULT_FILTERS: IFilter[] = FIXED_FREQUENCIES.map((f) => {
+  return {
+    frequency: f,
+    gain: 0,
+    quality: 1,
+    type: FilterTypeEnum.PEAK,
+  };
+});
+
+const DEFAULT_STATE: IState = {
+  isEnabled: true,
+  preAmp: 0,
+  filters: DEFAULT_FILTERS,
+};
+
+export const stateToString = (state: IState) => {
+  if (!state.isEnabled) {
+    return '';
+  }
+
+  let output: string[] = [];
+
+  output.push('Device: all');
+  output.push('Channel: all');
+
+  if (state.preAmp) {
+    output.push(`Preamp: ${state.preAmp}dB`);
```

Comment:
Would it not be safer to always push `Preamp: ${state.preAmp}dB`?
i.e. if preAmp = 0, push Preamp: 0dB

Reply:
Hmm, I guess it wouldn't hurt to add that. It does mean we have to write more content into the file each time though. @hemit-s @xenown do we have a strong preference for one or the other?

Reply:
I guess I'm concerned of what happens if equalizerAPO changes their default behaviour.

Reply:
Ill just add a comment. If the assumption that no preamp line implies 0db is false, someone presumably will notice that when we they set preamp to 0db, it sounds too loud / quiet.

Reply:
that's a fair point. We can add them in then - it'll also make it more clear what settings we're actually applying

Reply:
@xenown I'm fine with removing the if here if you want to (I actually already did in a previous commit - idk if you added it back on purpose or not?)

Reply:
oops, I guess I messed up my git stash conflict. Ill remove it.